e2e/testcases/cli_test.go

```diff
@@ -1276,3 +1285,311 @@ func TestApiResourceFormatting(t *testing.T) {
+			"git-importer", configsync.ControllerNamespace)
+	})
+	tg.Go(func() error {
+		return nt.Watcher.WatchForNotFound(kinds.CustomResourceDefinitionV1(),
+			util.ConfigManagementCRDName, "")
+	})
+	if err := tg.Wait(); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Wait for expected resources to be current...")
+	tg = taskgroup.New()
+	tg.Go(func() error {
+		return nt.Watcher.WatchForCurrentStatus(kinds.Deployment(),
+			util.ReconcilerManagerName, configsync.ControllerNamespace)
+	})
+	tg.Go(func() error {
+		return nt.Watcher.WatchForCurrentStatus(kinds.Deployment(),
+			configmanagement.RGControllerName, configmanagement.RGControllerNamespace)
+	})
+	tg.Go(func() error {
+		return nt.Watcher.WatchForNotFound(kinds.Deployment(),
+			core.RootReconcilerName(configsync.RootSyncName), configsync.ControllerNamespace)
+	})
+	tg.Go(func() error {
+		return nt.Watcher.WatchForNotFound(kinds.RootSyncV1Beta1(),
+			configsync.RootSyncName, configsync.ControllerNamespace)
+	})
+	if err := tg.Wait(); err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestNomosMigrateMonoRepo(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.NomosCLI, ntopts.SkipConfigSyncInstall)
+
+	nt.T.Cleanup(func() {
+		// Restore state of Config Sync installation after test.
+		// This also emulates upgrading to the current version after migrating
+		if err := nomostest.InstallConfigSync(nt); err != nil {
+			nt.T.Fatal(err)
+		}
+	})
+	nt.T.Cleanup(func() {
+		// Delete the ConfigManagement operator in case the test failed early.
+		// If this lingers around it could cause issues for subsequent tests.
+		cmDeployment := fake.DeploymentObject(
+			core.Namespace(configsync.ControllerNamespace),
+			core.Name("config-management"),
+		)
+		if err := nt.KubeClient.Delete(cmDeployment); err != nil && !apierrors.IsNotFound(err) {
+			nt.T.Fatal(err)
+		}
+		// Ensure the RootSync is deleted since it's not registered with nt
+		rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+		if err := nt.KubeClient.Delete(rs); err != nil && !apierrors.IsNotFound(err) {
+			nt.T.Fatal(err)
+		}
+	})
+
+	cmOperatorYaml := path.Join(nt.TmpDir, "config-management-operator.yaml")
+	out, err := nt.Shell.Command("gsutil", "cp",
+		"gs://config-management-release/released/1.18.0/config-management-operator.yaml",
+		cmOperatorYaml).CombinedOutput()
+	if err != nil {
+		nt.T.Error(string(out))
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Installing ConfigManagement")
+	nt.MustKubectl("apply", "-f", cmOperatorYaml)
+
+	if err := nt.Watcher.WatchForCurrentStatus(kinds.CustomResourceDefinitionV1(), util.ConfigManagementCRDName, ""); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	cmObj := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "configmanagement.gke.io/v1",
+			"kind":       "ConfigManagement",
+			"metadata": map[string]interface{}{
+				"name": "config-management",
+			},
+			"spec": map[string]interface{}{
+				"enableMultiRepo": true,
```

Comment:
We can start with monorepo mode initially.

Reply:
It's intentional to start with multi-repo mode here, because this causes the ACM operator to take ownership of the multi-repo objects. This means when it switches to mono-repo mode the multi-repo objects will be deleted.

If this starts with monorepo mode initially, then the multi-repo objects will still be installed alongside the mono-repo objects. The alternative is to write some code to manually clean up all of the multi-repo objects, but this seemed simpler.